test/unit/test_vector.c

```diff
@@ -61,6 +61,33 @@ static bool _init_test_pvector2(RPVector *v, size_t len, size_t padding) {
 	mu_assert ("init_test_pvector2", _r); \
 }
 
+static bool test_vector_fini() {
+	RVector v;
+	r_vector_init (&v, sizeof (void *), NULL, NULL);
+	r_vector_push (&v, &v);
+	mu_assert_eq_fmt (v.elem_size, 8UL, "init elem_size", "%lu");
+	mu_assert_eq_fmt (v.len, 1UL, "init len", "%lu");
+	mu_assert_eq_fmt ((void*)(size_t)!v.a, NULL, "init a", "%p");
```

Comment:
`mu_assert_notnull (v.a, "init a");`

Reply:
use `mu_assert_null`/`mu_assert_notnull` for all the other pointers operation.

Reply:
all those tests are not mine. but ill fix them

Reply:
yeah i wasn't looking at the others actually. just saw yours. they could have been better as well :/

Reply:
i just followed the style, but now its all fixed